src/main/scala/io/iohk/ethereum/extvm/VMClient.scala

```diff
@@ -14,13 +14,17 @@ import scala.annotation.tailrec
   *                 This is useful to override configuration for each test, rather than to recreate the VM.
   */
 class VMClient(
+    externalVmConfig: VmConfig.ExternalConfig,
     messageHandler: MessageHandler,
     testMode: Boolean)
   extends Logger {
 
   def sendHello(version: String, blockchainConfig: BlockchainConfig): Unit = {
     val config = BlockchainConfigForEvm(blockchainConfig)
-    val configMsg = msg.Hello.Config.EthereumConfig(buildEthereumConfigMsg(BlockchainConfigForEvm(blockchainConfig)))
+    val configMsg = externalVmConfig.vmType match {
+      case VmConfig.ExternalConfig.VmTypeIele => msg.Hello.Config.IeleConfig(buildIeleConfigMsg())
```

Comment:
based on our discussion on slack, weren't we planning for our near-term solution to be to make IeleConfig also be passed on every CallContext?